src/sentry/utils/snuba.py

```diff
@@ -234,7 +234,7 @@ def nest_groups(data, groups, aggregate_cols):
 
 
 def is_condition(cond_or_list):
-    return len(cond_or_list) == 3 and isinstance(cond_or_list[0], six.string_types)
+    return len(cond_or_list) == 3 and isinstance(cond_or_list[0], (six.string_types, list, tuple))
```

Comment:
I think this potentially misidentifies things as conditions. Seeing as we are using the convention:
```
"conditions": [[cond1, cond2], cond3, cond4]
```
to mean:
```
(cond1 OR cond2) AND cond3 AND cond4
```
this logic would incorrectly identify the whole set of conditions as a single condition

Reply:
I think the logic within snuba might actually have the same bug too.

Reply:
Well, whatchu recommend we do here? I'm not sure enough of what the real fix it, or what all is allowed.

But I'm pretty sure this is needed to support soemthing like, `[['a', 'b'], 'IN', [[1, 2]]]`

Reply:
Hmm, I guess first question is, do we need this type of `IN`? Like, could we get by with `a = 1 AND b = 2` instead of `(a, b) IN (1, 2)` ? 

If not, I will have to work through the syntax, as I think we have also potentially conflicts with the nested function call situation. (in which case you might have to use the `tuple(a, b)` function rather than the literal `(a, b)` tuple)

In either case, the only user of this `is_condition` is `flat_conditions` which is used to get a list of all the conditions to figure out if any of them refer to the `issue` column. Whatever we decide to do with the syntax I think that we need to make a change so that we can catch nested references to `issue` in these expressions.